### PR TITLE
Add dead link checker action

### PR DESCRIPTION
**Pull request by RemakingEden**
A pull request for issue https://github.com/commjoen/wrongsecrets/issues/302. I may need further permissions for this as I will need to test if it works correctly, there is also probably a discussion needed about exactly where it should check for dead links 

- [x] Add Github token as a secret  for repo so that Github requests dont hit rate limiter @commjoen I think this will need to be yourself. The variable I have used is "GITHUB_TOKEN". It needs a  personal token with no permissions.
- [x] Discuss exactly what should/shouldnt be included

### CODE REVIEW

**Comment by RemakingEden** [comment]
Okay so you can see the kind of output I would expect now from the latest run, I would intend on having this run on a cron job and allow it to make a new issue if it finds problems like dependabot does. This is the latest output which I think includes some valid finds

---

| Status        | Count |
|---------------|-------|
| 🔍 Total      | 175   |
| ✅ Successful | 148   |
| ⏳ Timeouts   | 0     |
| 🔀 Redirected | 0     |
| 👻 Excluded   | 24    |
| ❓ Unknown    | 0     |
| 🚫 Errors     | 3     |


## Errors per input
### Errors in aws/README.md
* [https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/managed_node_groups](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/managed_node_groups): Failed: Network error

### Errors in src/main/resources/templates/welcome.html
* [file:///github/workspace/src/main/resources/templates/spoil-7](file:///github/workspace/src/main/resources/templates/spoil-7): Failed: Cannot find file

### Errors in azure/README.md
* [https://azure.github.io/secrets-store-csi-driver-provider-azure/getting-started/usage/#provide-identity-to-access-key-vault](https://azure.github.io/secrets-store-csi-driver-provider-azure/getting-started/usage/#provide-identity-to-access-key-vault): Failed: Network error

**Comment by commjoen** [comment]
Nice one! At the MSTG project we used https://github.com/OWASP/owasp-mstg/blob/master/.github/workflows/checkLinks.yml which never ran into exhaustion as we used to run it on push to / pull request to master. Can we do the same here so that we don't need to do the token stuff? We have PRs/pushes to master about 3/week :) which is good enough to me :).

File: .github/workflows/link_checker.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+name: Dead Link Checker
+
+on:
+  repository_dispatch:
+  workflow_dispatch:
+  push:
+  pull_request:
+    branches: [ master ]
+  schedule:
```
Comment (commjoen) [inline .github/workflows/link_checker.yml:11]: can we just do 
- workflow,
- push to master (release check :D),
- PR ?

File: .github/workflows/link_checker.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1.4.1
+        with:
+          args: --exclude-all-private -r 2 './**/*.md' './**/*.html'
+          fail: true
+        env:
```
Comment (commjoen) [inline .github/workflows/link_checker.yml:26]: With the schedule set,maybe we don't need a PAT then :)

**Review by commjoen — CHANGES_REQUESTED** [review]
Love this! 2 minor comments+ request: would you be able to help me please?
- can youfix the broken links?
- is the .lycheeignore configurable in terms of its location? it would be nice to centralize all config :).

File: .github/workflows/link_checker.yml (hunk 2 above)
Comment (RemakingEden) [reply .github/workflows/link_checker.yml:26]: Shall we try the schedule without any others and see if that hits the rate limit? I guess it depends on what has been done on that Github shared runner. If this doesnt work we can try something different

**Comment by RemakingEden** [comment]
> Love this! 2 minor comments+ request: would you be able to help me please?
> 
>     * can youfix the broken links?
> 
>     * is the .lycheeignore configurable in terms of its location? it would be nice to centralize all config :).

@commjoen  Okay .lycheeconfig is now in the config folder. Is that alright?

As for fixing links i have fixed 2 of the 3. I'm going to paste one below and I'll need you to fix that ideally as I don't know what it should be replaced with.

Errors in azure/README.md

    https://azure.github.io/secrets-store-csi-driver-provider-azure/getting-started/usage/#provide-identity-to-access-key-vault: Failed: Network error

**Comment by commjoen** [comment]
This is awesome! Thanks! (Will dive in later this week ;-).

**Review by commjoen — APPROVED** [review]
Thanks!

**Pull request merged**